### ISSUE TITLE
darwin.apple_sdk.sdk doesn't exist on aarch64

### ISSUE THREAD

**Issue by angustrau** [NixOS/nixpkgs#131791]
### Describe the bug
The `sdk` attribute on `darwin.apple_sdk` doesn't exist on `aarch64-darwin`. I've tracked down this difference to [this line](https://github.com/NixOS/nixpkgs/blob/f4ce33bf06b56bf5f7792127d1accf8823d30bd8/pkgs/top-level/darwin-packages.nix#L45) in `pkgs/top-level/darwin-packages.nix`, which specifies the use of a different Apple SDK version on aarch64. [This version](https://github.com/NixOS/nixpkgs/blob/a66d9f9b000f7ddf07d0a5ed1d729a1f36c179c6/pkgs/os-specific/darwin/apple-sdk-11.0/default.nix#L45) of the Apple SDK derivation doesn't have an `sdk` attribute, instead defining `darwin.apple_sdk.MacOSX-SDK`. `MacOSX-SDK` is not quite a drop in replacement for `sdk`, as the derivation has a slightly different file structure. I was able to hack a proof of concept fix by applying the following changes

```diff
diff --git a/pkgs/os-specific/darwin/apple-sdk-11.0/default.nix b/pkgs/os-specific/darwin/apple-sdk-11.0/default.nix
index 44b119e1a23..1e476b92781 100644
--- a/pkgs/os-specific/darwin/apple-sdk-11.0/default.nix
+++ b/pkgs/os-specific/darwin/apple-sdk-11.0/default.nix
@@ -28,6 +28,7 @@ let
 
       mkdir $out
       cp -r System usr $out/
+      cp -r usr/* $out
     '';
 
     passthru = {
@@ -43,6 +44,7 @@ let
     # TODO: this is nice to be private. is it worth the callPackage above?
     # Probably, I don't think that callPackage costs much at all.
     inherit MacOSX-SDK;
+    sdk = MacOSX-SDK;
 
     Libsystem = callPackage ./libSystem.nix {};
```

This fix allowed nix to at least get past the evaluation stage. As the Apple SDK derivation is a low-level dependency, it will take a while to recompile on my machine and test that it is completely successful. I will make sure to update as soon as it is done.

### Steps To Reproduce
1. Checkout master
2. `export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1 ` to force `aften` to evaluate (or checkout #131666)
3. `nix-build . -A webkitgtk`

Result:
```
error: attribute 'sdk' missing

       at /Users/angustrau/Documents/git/nixpkgs-2/pkgs/top-level/all-packages.nix:18902:63:

        18901|
        18902|   webkitgtk = callPackage ../development/libraries/webkitgtk {
             |                                                               ^
        18903|     harfbuzz = harfbuzzFull;
(use '--show-trace' to show detailed location information)
```

### Expected behavior
This is expected to evaluate successfully.

### Additional context
I first ran into this issue while debugging a failing test on #131666, where `webkitgtk` was failing to evaluate. (`aften` is a transitive dependency of `webkitgtk`). From a quick grep `webkitgtk` appears to be the only [user  of `darwin.apple_sdk.sdk`](https://github.com/NixOS/nixpkgs/blob/a66d9f9b000f7ddf07d0a5ed1d729a1f36c179c6/pkgs/development/libraries/webkitgtk/default.nix#L180).

### Notify maintainers

@thefloweringash

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 20.6.0, macOS 11.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4pre20210707_02dd6bb`
 - channels(angustrau): `""`
 - channels(root): `"nixpkgs-21.11pre303315.16105403bdd"`
 - nixpkgs: `/etc/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- darwin.apple_sdk
- webkitgtk
# a list of nixos modules affected by the problem
module:
```

**Comment by angustrau** [comment]
Update: With the patch it was able to recompile the toolchain (llvm, clang, etc) and got all the way up to compiling library dependencies before it failed. There's clearly more work to be done before `webkitgtk` is up and running, but it looks like this is on the right track!

```
ld: file not found: /nix/store/qmql2qgm802z6w0vhajjqd3dwrkviihh-clang-wrapper-11.1.0/resource-root/lib/darwin/libclang_rt.ubsan_osx_dynamic.dylib
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [gme/CMakeFiles/gme.dir/build.make:778: gme/libgme.0.6.3.dylib] Error 1
make[1]: *** [CMakeFiles/Makefile2:153: gme/CMakeFiles/gme.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
error: builder for '/nix/store/xfl1xbv6g04ffq6x378d2dhjgqg2frpc-libgme-0.6.3.drv' failed with exit code 2;
       last 10 log lines:
       > clang-11: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
       > 16 warnings generated.
       > 1 warning generated.
       > 16 warnings generated.
       > [100%] Linking CXX shared library libgme.dylib
       > ld: file not found: /nix/store/qmql2qgm802z6w0vhajjqd3dwrkviihh-clang-wrapper-11.1.0/resource-root/lib/darwin/libclang_rt.ubsan_osx_dynamic.dylib
       > clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
       > make[2]: *** [gme/CMakeFiles/gme.dir/build.make:778: gme/libgme.0.6.3.dylib] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:153: gme/CMakeFiles/gme.dir/all] Error 2
       > make: *** [Makefile:149: all] Error 2
       For full logs, run 'nix log /nix/store/xfl1xbv6g04ffq6x378d2dhjgqg2frpc-libgme-0.6.3.drv'.
error: 1 dependencies of derivation '/nix/store/rhyisnyni6naq0ww3yxpij7g4jwc6cc0-gst-plugins-bad-1.18.2.drv' failed to build
error (ignored): error: cannot unlink '/private/tmp/nix-build-curl-7.76.1.drv-0/curl-7.76.1': Directory not empty
error (ignored): error: cannot unlink '/private/tmp/nix-build-llvm-12.0.1.drv-1/llvm-12.0.1.src/test': Directory not empty
error: 1 dependencies of derivation '/nix/store/jni7q096z84iyy68n1wij0435x257fg4-webkitgtk-2.32.3.drv' failed to build
```

**Comment by thefloweringash** [comment]
In general the sdk is not meant to be used outside of the darwin packages. For an extra library that's required by a package, we can make this into a proper part of the interface (like in #116528). Though in this case the solution isn't quite as obvious, since the plucked header doesn't seem to correspond to a particular library.

**Comment by angustrau** [comment]
The header was introduced into webkitgtk in #123298. @veprbl would you have any insight into whether it should be pulled into its own package?

**Comment by veprbl** [comment]
My guess would be that libproc.h should be a part of the `xnu` package, but for some reason at the time the only usable version of it was in the `sdk`.

**Comment by angustrau** [comment]
I've found libproc.h is available in `darwin.usr-include`. It appears to be the same version as in `sdk`, but I've only checked on aarch64.

**Comment by angustrau** [comment]
I got webkitgtk to build successfully with this [series of patches](https://github.com/angustrau/nixpkgs/commits/webkit-aarch64-darwin)! I'm pretty new to nix and macOS in general, so I would love a review and some advice on how I can split this up into PRs.

**Comment by happysalada** [comment]
Some of those commits seem to be related to a single package. You could make separate PRs for this and get those isolated changes merged pretty quickly.

For the more involved changes, it would be good if you could get a review from someone with a Aarch64 macos.
If you look into the nixpkgs commit done for Aarch64 and ask some of the people who did those, you should be able to get feedback.

Let me know if anything doesn't make sense.

**Comment by angustrau** [comment]
@happysalada All of these changes have been PR'd and most have been accepted! The currently outstanding ones are #132273 and #131666. For webkitgtk I'm waiting till I can dig out an old intel mac to check the compilation issue, and aften is just blocked on webkitgtk.

**Comment by happysalada** [comment]
Amazing!

**Comment by angustrau** [comment]
Closed by #132273 after removing the use of `darwin.apple_sdk.sdk` on Apple Silicon
